apps/builder/app/canvas/features/webstudio-component/webstudio-component.tsx

```diff
@@ -142,6 +143,20 @@ export const WebstudioComponentDev = ({
     }
   }, [isSelected]);
 
+  // this assumes presence of `useStore(selectedInstanceSelectorStore)` above
+  // we rely on root re-rendering after selected instance changes
+  useEffect(() => {
+    // 1 means root
+    if (instanceSelector.length === 1) {
+      // If by the time root is rendered,
+      // no selected instance renders and sets `selectedInstanceIsRendered` to `true`,
+      // then it's clear that selected instance will not render at all, so we set it to `false`
+      if (selectedInstanceIsRenderedStore.get() === undefined) {
+        selectedInstanceIsRenderedStore.set(false);
+      }
+    }
+  });
```

Comment:
this is executed every single time, shouldn't it rather have dependencies, e.g. instanceSelector

Reply:
The code inside is very light. Checking dependencies can be even slower

Reply:
the danger is that someone can add more to the code without realizing it's going to run every time